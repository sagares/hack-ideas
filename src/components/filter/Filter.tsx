import React from "react";
import { FilterProps } from "../../HackIdea.types";
import { SearchBar } from ".";
import { SortMenu } from "./SortMenu";

import "./filter.less";

const Filter = ({ searchText, onFilter, sortBy, onSort }: FilterProps) => {
  return (
    <div className="hack-ideas--filter row">
      <div className="search-bar col s12 m12 l4">
        <SearchBar searchText={searchText} onFilter={onFilter} />
      </div>
      <div className="filter-dropdown col s4 m2 l2">
        <SortMenu sortBy={sortBy} onSort={onSort} />
      </div>
    </div>
  );
};

export default Filter;
