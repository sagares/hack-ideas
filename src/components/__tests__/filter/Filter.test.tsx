/**
 * @jest-environment jsdom
 */
import React from "react";
import { render } from "@testing-library/react";
import Filter from "../../filter";

describe("Filter Component", () => {
  let element, component, onFilterSpy, onSortSpy;

  beforeEach(() => {
    onFilterSpy = jest.fn();
    onSortSpy = jest.fn();

    element = document.createElement("div");

    component = render(
      <Filter
        searchText="mock-search"
        sortBy=""
        onFilter={onFilterSpy}
        onSort={onSortSpy}
      />,
      {
        container: document.body.appendChild(element),
      }
    );
  });

  afterEach(() => {
    document.body.removeChild(element);
  });

  test("renders", () => {
    expect(component.container).toMatchSnapshot();
  })
});
