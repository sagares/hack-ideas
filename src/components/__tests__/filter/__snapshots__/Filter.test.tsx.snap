// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component renders 1`] = `
<div>
  <div
    class="hack-ideas--filter row"
  >
    <div
      class="search-bar col s12 m12 l4"
    >
      <nav>
        <div
          class="nav-wrapper"
        >
          <form>
            <div
              class="input-field grey lighten-4"
            >
              <input
                id="search"
                type="search"
                value="mock-search"
              />
              <label
                class="label-icon"
                for="search"
              >
                <i
                  class="material-icons"
                >
                  search
                </i>
              </label>
              <i
                class="clear-icon material-icons"
              >
                close
              </i>
            </div>
          </form>
        </div>
      </nav>
    </div>
    <div
      class="filter-dropdown col s4 m2 l2"
    >
      <a
        class="dropdown-trigger btn-floating btn-large"
        data-target="dropdown1"
      >
        <i
          class="large material-icons"
        >
          sort
        </i>
      </a>
      <ul
        class="dropdown-content"
        id="dropdown1"
        tabindex="0"
      >
        <li
          class=""
          tabindex="0"
        >
          <a>
            <i
              class="material-icons"
            >
              access_time
            </i>
            By Created On
          </a>
        </li>
        <li
          class=""
          tabindex="0"
        >
          <a>
            <i
              class="material-icons"
            >
              thumb_up
            </i>
            By Upvotes
          </a>
        </li>
        <li
          class="divider"
          tabindex="-1"
        />
        <li
          tabindex="0"
        >
          <a>
            <i
              class="material-icons"
            >
              cached
            </i>
            Reset
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
